Core/GameEngine/Source/Common/System/MiniDumper.cpp

```diff
@@ -0,0 +1,682 @@
+	{
+		DEBUG_LOG(("MiniDumper::MiniDumpCallback: Required parameters were null; CallbackParam=%p, CallbackInput=%p, CallbackOutput=%p.",
+			CallbackParam, CallbackInput, CallbackOutput));
+		return false;
+	}
+
+	MiniDumper* dumper = static_cast<MiniDumper*>(CallbackParam);
+	return dumper->CallbackInternal(*CallbackInput, *CallbackOutput);
+}
+
+Bool MiniDumper::ShouldWriteDataSegsForModule(const PWCHAR module) const
+{
+	// Only include data segments for the game, ntdll and kernel32 modules to keep dump size low
+	static constexpr const WideChar* wanted_modules[] = { L"ntdll.dll", L"kernel32.dll"};
+	if (endsWithNoCase(module, m_executablePath))
+	{
+		return true;
+	}
+
+	for (size_t i = 0; i < ARRAY_SIZE(wanted_modules); ++i)
+	{
+		if (endsWithNoCase(module, wanted_modules[i]))
+		{
+			return true;
+		}
+	}
+
+	return false;
+}
+
+// This is where the memory regions and things are being filtered
+BOOL MiniDumper::CallbackInternal(const MINIDUMP_CALLBACK_INPUT& input, MINIDUMP_CALLBACK_OUTPUT& output)
+{
+	BOOL success = TRUE;
+	switch (input.CallbackType)
+	{
+	case IncludeModuleCallback:
+	case ThreadCallback:
+	case ThreadExCallback:
+		break;
+	case ModuleCallback:
+		if (output.ModuleWriteFlags & ModuleWriteDataSeg)
+		{
+			if (!ShouldWriteDataSegsForModule(input.Module.FullPath))
+			{
+				// Exclude data segments for the module
+				output.ModuleWriteFlags &= (~ModuleWriteDataSeg);
+			}
+		}
+		break;
+	case IncludeThreadCallback:
+		// We want all threads except the dumping thread
+		if (input.IncludeThread.ThreadId == m_dumpThreadId)
+		{
+			output.ThreadWriteFlags &= (~ThreadWriteThread);
+		}
+		break;
+	case MemoryCallback:
+#ifndef DISABLE_GAMEMEMORY
+		// DumpMemoryObjects will set output.MemorySize to 0 once it's completed,
+		// signalling to end the memory callbacks.
+		DumpMemoryObjects(output.MemoryBase, output.MemorySize);
+#else
+		output.MemoryBase = 0;
+		output.MemorySize = 0;
+#endif
+		break;
+	case ReadMemoryFailureCallback:
+		DEBUG_LOG(("MiniDumper::CallbackInternal: ReadMemoryFailure with MemoryBase=%llu, MemorySize=%lu: error=%u",
+			input.ReadMemoryFailure.Offset, input.ReadMemoryFailure.Bytes, input.ReadMemoryFailure.FailureStatus));
+		break;
+	case CancelCallback:
+		// The cancel callback checks if the current minidump should be cancelled.
+		// Set output to never wish to cancel, nor to receive cancel callbacks in the future.
+		output.Cancel = FALSE;
+		output.CheckCancel = FALSE;
+		break;
+	}
+
+	return success;
+}
+
+#ifndef DISABLE_GAMEMEMORY
+void MiniDumper::MoveToNextAllocatorWithRawBlocks()
+{
+	// Skip over DMAs that have no raw blocks
+	while (m_currentAllocator && !m_currentAllocator->getFirstRawBlock())
+	{
+		m_currentAllocator = m_currentAllocator->getNextDmaInList();
+	}
+
+	if (m_currentAllocator)
+	{
+		m_currentSingleBlock = m_currentAllocator->getFirstRawBlock();
+	}
+}
+
+void MiniDumper::MoveToNextSingleBlock()
```

Comment:
`MoveToNextAllocatorWithRawBlocks` and `MoveToNextSingleBlock` can likely be one function like how you did it in the Memory Pool Iterator.